packages/autorest.python/autorest/jsonrpc/server.py

```diff
@@ -31,6 +32,7 @@ def GetPluginNames():
 
 
 @dispatcher.add_method
+@no_type_check
```

Comment:
what is this decorator, and do we really want to use it?

Reply:
Here I add `typing.no_type_check` to ignore type errors on the `Process()` function. 

I'm doing this because the `pyright` and `mypy` have contradictory results on below codes: https://github.com/Azure/autorest.python/blob/5d50efab6887e830f72d86916eb360fcc4b36983/packages/autorest.python/autorest/jsonrpc/server.py#L46-L61.
The pyright report ` error: Unnecessary "# type: ignore" comment` on these `type: ignore`s.
However, if I delete these `type: ignore`s, the mypy will report `error: Name "PluginToLoad" already defined (possibly by an import)`

Reply:
@changlong-liu were there any other incidences of pyright and mypy disagreeing?

Reply:
Yes @kristapratico , another case pyright disagree with mypy is:
For `# type: ignore` at below, pyright report it's an unnecessary ignore, but mypy report below if we don't have it. https://github.com/Azure/autorest.python/blob/148737f1ce70e22525d408f97a59635c096968df/packages/autorest.python/test/azure/legacy/Expected/AcceptanceTests/Lro/lro/aio/operations/_lr_os_custom_header_operations.py#L589

For this case, I'm adding `# pyright: reportUnnecessaryTypeIgnoreComment=false` in the related file.

Reply:
Thanks. Because mypy and pyright disagree enough for it to be annoying, I think I will change both reportUnnecessaryTypeIgnoreComment and reportUnnecessaryCast from "true" to "warning" in the pyrightconfig.json in the python repo. Feel free to do the same in autorest.python.

Reply:
sure @kristapratico , I don't find disagreements on `reportUnnecessaryCast`, but `reportUnnecessaryTypeIgnoreComment` does make some confusing cases.

Reply:
removed this logic since we are now not using `reportUnnecessaryTypeIgnoreComment`